[8.x] Remove invalid Blade compiler test

jrmajor opened the pull request:
I'm not sure what is `ViewBladeCompilerTest::testDontIncludeNullPath()` meant to test, but within this test `null` is passed to `BladeCompiler::setPath()`, which docblock says it only accepts strings. It's a setter for `BladeCompiler::$path` which, according to docblock, also isn't nullable.

https://github.com/laravel/framework/blob/9267b0c4b2ea31b82f45715d047d76657635152e/tests/View/ViewBladeCompilerTest.php#L186

~~Unless I'm missing something, I'll send a PR to 9.x to remove this test and revert changes in `BladeCompiler`. If you decide this is the desired behavior, I'll update this PR to change types in docblocks to `?string`.~~

jrmajor commented on tests/View/ViewBladeCompilerTest.php:
```diff
@@ -182,7 +182,7 @@ public function testDontIncludeNullPath()
         $files->shouldReceive('exists')->once()->with(__DIR__)->andReturn(true);
-        $files->shouldReceive('put')->once()->with(__DIR__.'/'.sha1(null).'.php', 'Hello World');
+        $files->shouldReceive('put')->once()->with(__DIR__.'/'.sha1('').'.php', 'Hello World');
         $compiler->setPath(null);
```
`BladeCompiler::setPath()` signature:

```php
/**
 * Set the path currently being compiled.
 *
 * @param  string  $path
 * @return void
 */
public function setPath($path)
```

`BladeCompiler::$path`:

```php
/**
 * The file currently being compiled.
 *
 * @var string
 */
protected $path;
```

GrahamCampbell commented on src/Illuminate/View/Compilers/BladeCompiler.php:
```diff
@@ -211,7 +211,7 @@ public function getPath()
     public function setPath($path)
     {
-        $this->path = $path;
+        $this->path = $path ?? '';
```
This looks wrong. The phpdoc says you must pass a string. Any code that calls this with null is wrong and needs to be fixed.

jrmajor replied:
Should this test just be removed, then?

selcukcukur replied:
I think it's not a good idea to send it as a string when it is empty.  And I don't find it right to make these changes before this 8.1 is released. All together, we should see what happens when that version arrives and act that way, we may be making a lot of changes unnecessarily now.

jrmajor replied:
@selcukcukur 

> And I don't find it right to make these changes before this 8.1 is released.

Thankfully the maintainers do.

> we may be making a lot of changes unnecessarily now

I've provided you with links to accepted RFC in reply to your comments on my previous PR.

jrmajor committed:
remove invalid test

jrmajor commented:
I've updated this PR to just remove the invalid test (no point changing null to an empty string in this one, there's another test for that).

GrahamCampbell commented on tests/View/ViewBladeCompilerTest.php:
```diff
@@ -177,16 +177,6 @@ public function testDontIncludeEmptyPath()
-        $files->shouldReceive('get')->once()->with(null)->andReturn('Hello World');
-        $files->shouldReceive('exists')->once()->with(__DIR__)->andReturn(true);
-        $files->shouldReceive('put')->once()->with(__DIR__.'/'.sha1(null).'.php', 'Hello World');
-        $compiler->setPath(null);
```
Please find all places in the framework that might call this will null, and fix them.

GrahamCampbell replied:
There is at least one, in the service provider.

jrmajor replied:
<img width="689" alt="image" src="https://user-images.githubusercontent.com/26096713/115970209-41c7c300-a541-11eb-9d28-490fe2dc1aac.png">

jrmajor replied:
It's OK.

jrmajor replied:
> There is at least one, in the service provider.

Sorry, didn't see this comment. Where exactly it is? Even using normal search, without any PhpStorm magic, I can't find it.

taylorotwell commented:
I don't see any reason to remove the test at the moment?

jrmajor commented:
@taylorotwell This test passes `null` to `BladeCompiler::setPath()`, which accepts only a string, **causing this test to fail on PHP 8.1**. In the first version of this PR I've just added `?? ''` in this method to make sure that `BladeCompiler::$path` is always a string, but @GrahamCampbell told me, that “any code that calls this with null is wrong and needs to be fixed”. Fixing this test would mean changing `null` to `''`, but there's another test for that, so I've removed this one.
